web-api/src/storage/memory.rs

```diff
@@ -407,6 +384,49 @@ impl storage::Interaction for Storage {
 
         Ok(())
     }
+
+    async fn update_interactions<F>(
+        &self,
+        user_id: &UserId,
+        updated_document_ids: &[&DocumentId],
+        mut update_logic: F,
+    ) -> Result<(), Error>
+    where
+        F: for<'a, 'b> FnMut(InteractionUpdateContext<'a, 'b>) -> &'b PositiveCoi + Send + Sync,
+    {
+        // This doesn't have the exact same concurrency semantics as the postgres version
```

Comment:
The memory storage is to speed up some tests and to not have to bootup the whole system. Nowhere we require that it should mimic every aspect of postgres.

Reply:
Yes, but it can mean that tests might fail with the memory storage but not postgres or the other way around, or that behavior observed with the memory storage (e.g. improvements of score) only apply the the memory storage as they subtle depend on this differences.